src/backend/distributed/utils/maintenanced.c

```diff
@@ -298,6 +298,16 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		double timeout = 10000.0; /* use this if the deadlock detection is disabled */
 		bool foundDeadlock = false;
 
+		LWLockAcquire(&MaintenanceDaemonControl->lock, LW_EXCLUSIVE);
```

Comment:
Do have some worry that `StopMaintenanceDaemon` has a race condition where between HASH_REMOVE & us actually dying we're incorrect. That's an existing issue. As for this check, it could be proactively checked when we're setting workerPid to MyProcPid & then calling `kill` from the new worker, I happened to go this route

Reply:
It may also make sense to drop the lock/unlock. Even if we fix the root of the issue, we should have an `Assert(myDbData->workerPid == MyProcPid)`